# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1208291452 14400
#      Tue Apr 15 16:30:52 2008 -0400
# Branch release-3-0-x
# Node ID ee21531f1b5e8e7ea1e51bc267e252bf14bc4634
# Parent  e474c782e1e2529d3ac181bead28e69b5fafa67e
view: get values from current axes if nargin == 0

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-04-15  John W. Eaton  <jwe@octave.org>
+
+	* plot/view.m: Get values from current axes if nargin == 0.
+
 2008-04-09  John W. Eaton  <jwe@octave.org>
 
 	* deprecated/beta_cdf.m, deprecated/beta_inv.m,
 	deprecated/beta_pdf.m, deprecated/beta_rnd.m,
 	deprecated/binomial_cdf.m, deprecated/binomial_inv.m,
 	deprecated/binomial_pdf.m, deprecated/binomial_rnd.m,
 	deprecated/chisquare_cdf.m, deprecated/chisquare_inv.m,
 	deprecated/chisquare_pdf.m, deprecated/chisquare_rnd.m,
diff --git a/scripts/plot/view.m b/scripts/plot/view.m
--- a/scripts/plot/view.m
+++ b/scripts/plot/view.m
@@ -23,17 +23,21 @@
 ## Set or get the viewpoint for the current axes.
 ## @end deftypefn
 
 ## Author: jwe
 
 function [azimuth, elevation] = view (x, y, z)
 
   if (nargin < 4)
-    if (nargin == 1)
+    if (nargin == 0)
+      tmp = get (gca (), "view");
+      az = tmp(1);
+      el = tmp(2);
+    elseif (nargin == 1)
       if (x == 2)
 	az = 0;
 	el = 90;
       elseif (x == 3)
 	az = -37.5;
 	el = 30;
       else
 	error ("view: expecting single argument to be 2 or 3");
